friendshipper/src/lib/components/preferences/PreferencesModal.svelte

```diff
@@ -201,15 +217,53 @@
 				await emit('error', e);
 			}
 
+			let playtestPromise: Nullable<Promise> = null;
+			let statusPromise: Nullable<Promise> = null;
+			let commitsPromise: Nullable<Promise> = null;
+			let workflowsPromise: Nullable<Promise> = null;
+
 			if (regionChanged) {
-				$playtests = await getPlaytests();
+				playtestPromise = getPlaytests();
+			}
+
+			if (hasRepoUrlChanged) {
+				statusPromise = getRepoStatus();
+				commitsPromise = getAllCommits();
+
+				if (localAppConfig.repoUrl !== '') {
+					workflowsPromise = await getWorkflows();
+				}
+			}
+
+			const promises: Promise[] = [
+				playtestPromise,
+				statusPromise,
+				commitsPromise,
+				workflowsPromise
+			];
+			const values = await Promise.all(promises);
```

Comment:
for your daily dose of typescript nonsense, you can do this like this:

```ts
const { playtestResponse, statusResponse, commitsResponse, workflowsResponse } = await Promise.all([playtestPromise, statusPromise, commitsPromise, workflowsPromise])
```

if you'd like to avoid the hard index checking